(1) Community Patch/Core Files/Overrides/Includes/IconSupport.lua

```diff
@@ -0,0 +1,256 @@
+		end
+	end
+
+	return true;
+end
+
+---------------------------------------------------------------------------------------------------------
+-- Art Team Color Variables
+---------------------------------------------------------------------------------------------------------
+local teamSize80 = 64;
+local teamSize64 = 48;
+local teamSize45 = 32;
+local teamSize32 = 24;
+---------------------------------------------------------------------------------------------------------
+-- This is a special case hookup for civilization icons that will take into account
+-- the fact that player colors are dynamically handed out
+---------------------------------------------------------------------------------------------------------
+function CivIconHookup(playerID, iconSize, iconControl, teamColorControl, shadowIconControl, alwaysUseComposite, shadowed,
+					   highlightControl)
+	local thisCivType = PreGame.GetCivilization(playerID);
+	local thisCiv = GameInfo.Civilizations[thisCivType];
+	local thisPlayerColorIndex = nil;
+	local defaultColorSet = nil;
+	local nonDefaultColor = nil;
+
+	if highlightControl ~= nil then
+		highlightControl:SetHide(true);
+	end
+
+	if thisCiv then
+		thisPlayerColorIndex = PreGame.GetCivilizationColor(playerID);
+		defaultColorSet = GameInfo.PlayerColors[thisCiv.DefaultPlayerColor];
+		nonDefaultColor = (defaultColorSet.ID ~= thisPlayerColorIndex);
+		if playerID >= GameDefines.MAX_MAJOR_CIVS then
+			nonDefaultColor = true;
+		end
+	end
+
+	if thisCiv and (alwaysUseComposite or nonDefaultColor) then
+		if (iconSize == 80) then
+			iconSize = teamSize80;
+		elseif (iconSize == 64) then
+			iconSize = teamSize64;
+		elseif (iconSize == 45) then
+			iconSize = teamSize45;
+		elseif (iconSize == 32) then
+			iconSize = teamSize32;
+		end
+
+		local textureOffset, textureAtlas = IconLookup(thisCiv.PortraitIndex, iconSize, thisCiv.AlphaIconAtlas);
+		if textureOffset ~= nil then
+			iconControl:SetHide(false);
+			iconControl:SetTexture(textureAtlas);
+			iconControl:SetTextureOffset(textureOffset);
+			if shadowIconControl and shadowed then
+				shadowIconControl:SetTexture(textureAtlas);
+				shadowIconControl:SetTextureOffset(textureOffset);
+			end
+		end
+
+		local thisPlayerColorSet;
+		if thisPlayerColorIndex > -1 then
+			thisPlayerColorSet = GameInfo.PlayerColors[thisPlayerColorIndex];
+		else
+			thisPlayerColorSet = defaultColorSet;
+		end
+
+		local primaryColor         = GameInfo.Colors[thisPlayerColorSet.PrimaryColor];
+		local primaryColorVector   = Vector4(primaryColor.Red, primaryColor.Green, primaryColor.Blue, primaryColor.Alpha);
+		local secondaryColor       = GameInfo.Colors[thisPlayerColorSet.SecondaryColor];
+		local secondaryColorVector = Vector4(secondaryColor.Red, secondaryColor.Green, secondaryColor.Blue,
+			secondaryColor.Alpha);
+
+		iconControl:SetColor(primaryColorVector);
+
+		if teamColorControl then
+			if (iconSize == teamSize80) then
```

Comment:
The only difference is offset x, use table lookup and do teamColorControl:SetTexture("CivIconBGSizes.dds") only once